dirsrvtests/tests/suites/plugins/managed_entry_test.py

```diff
@@ -36,14 +40,74 @@ def _create_inital(topo):
     conf_mep.create(properties={'cn': 'UPG Definition1', 'originScope': f'cn=Users,{DEFAULT_SUFFIX}',
-                                             'originFilter': 'objectclass=posixaccount',
-                                             'managedBase': f'cn=Groups,{DEFAULT_SUFFIX}',
-                                             'managedTemplate': mep_template1.dn})
+                                'originFilter': 'objectclass=posixaccount', 'managedBase': f'cn=Groups,{DEFAULT_SUFFIX}',
+                                'managedTemplate': mep_template1.dn})
     container = nsContainers(topo.standalone, DEFAULT_SUFFIX)
     for cn in ['Users', 'Groups']:
         container.create(properties={'cn': cn})
 
 
+def test_managed_entry_betxn(topo):
+    """Test if failure to create a managed entry rolls back the transaction.
+
+    :id: 7aa74994-f89b-11ec-9821-98fa9ba19b65
+    :setup: Standalone Instance
+    :steps:
+        1. Check that plugin active if not activate it
+        2. Create a Template entry
+        3. Create a definition entry
+        4. Attempt to create a user
+        5. Verify that transaction is aborted and user not created
+    :expected results:
+        1. Success
+        2. Success
+        3. Success
+        4. Success
+        5. Success
+    """
+
+    log.info("Make sure the plugin is active")
+    me_plugn = ManagedEntriesPlugin(topo.standalone)
+    log.info("Stopping and starting the Managed Entry plugin.")
+    me_plugn.disable()
+    me_plugn.enable()
+    assert me_plugn.status()
+    log.info("Plugin Restarted.")
+    log.info("Adding organization units")
+    ous = OrganizationalUnits(topo.standalone, DEFAULT_SUFFIX)
+    ou_people = ous.create(properties={'ou': 'managed_people'})
+    ou_groups = ous.create(properties={'ou': 'managed_groups'})
+
+    log.info("Create the template entry")
+    mep_templates = MEPTemplates(topo.standalone, DEFAULT_SUFFIX)
+    mep_temp1 = mep_templates.create(properties={
+        'cn': 'MEP template',
+        'mepRDNAttr': 'cn',
+        'mepStaticAttr': 'objectclass: groupOfNames|objectclass: extensibleObject'.split('|'),
+        'mepMappedAttr': 'cn: $cn|uid: $cn|gidNumber: $uidNumber'.split('|')
+    })
+    conf_mep = MEPConfigs(topo.standalone)
+    log.info("Create definition entry.")
+    conf_mep.create(properties={
+        'cn': 'cn=config',
+        'originScope': ou_people.dn,
+        'originFilter': 'objectclass=posixAccount',
+        'managedBase': ou_groups.dn,
+        'managedTemplate': mep_temp1.dn})
+    topo.standalone.restart()
+    log.info("Attempt to add a user that doenst fit the template.")
+    user = UserAccounts(topo.standalone, DEFAULT_SUFFIX, rdn='ou={}'.format(ou_people.rdn))
+    with pytest.raises(ldap.UNWILLING_TO_PERFORM):
+        mgd_entry = user.create(properties={
+            'uid': 'test_uid',
+            'cn': 'test_uid',
+            'sn': 'test_sn',
+        })
+    log.info("Cleaning up")
+    user.remove()
```

Comment:
Another issue I encounter is here:

```
AttributeError: 'UserAccounts' object has no attribute 'remove'
```

I think it should be `mgd_entry.delete` and, it probably should be run after the use is wrongly created in `with pytest.raises(ldap.UNWILLING_TO_PERFORM):` (because if everything is correct, the user is not created)

Reply:
my mistake will fix